configure.ac

```diff
@@ -361,15 +361,25 @@ AC_SUBST([GIRTYPELIBDIR])
 AM_CONDITIONAL([ENABLE_INTROSPECTION],[test "$enable_introspection" = "yes"])
 
 dnl ================== libsynctex ===========================================
+AC_ARG_ENABLE([synctex],
+  [AS_HELP_STRING([--disable-synctex], [Disable support for synctex])],
+  [],
+  [enable_synctex=yes])
 
-PKG_CHECK_MODULES(SYNCTEX, [synctex >= $SYNCTEX_REQUIRED], has_synctex=yes, has_synctex=no)
 
-dnl not found? use internal code copy.
-if test "x$has_synctex" = "xno"; then
-        AC_MSG_ERROR("SyncTeX support is disabled since library version $SYNCTEX_REQUIRED or newer not found")
+if test "$enable_synctex" = "yes"; then
+  AC_DEFINE([ENABLE_SYNCTEX],[1],[Define if synctex support is enabled])
+  PKG_CHECK_MODULES(SYNCTEX, [synctex >= $SYNCTEX_REQUIRED], has_synctex=yes, has_synctex=no)
+  dnl not found? use internal code copy.
+  if test "x$has_synctex" = "xno"; then
+          AC_MSG_ERROR("SyncTeX support is disabled since library version $SYNCTEX_REQUIRED or newer not found")
+  fi
+  AC_SUBST(SYNCTEX_LIBS)
+  AC_SUBST(SYNCTEX_CFLAGS)
 fi
-AC_SUBST(SYNCTEX_LIBS)
-AC_SUBST(SYNCTEX_CFLAGS)
+
+AM_CONDITIONAL([ENABLE_SYNCTEX], [test "$enable_synctex" = "yes"])
```

Comment:
The ENABLE_SYNCTEX conditional is defined but it's not used.
```diff
--- a/libdocument/Makefile.am
+++ b/libdocument/Makefile.am
@@ -102,11 +102,14 @@ libatrildocument_la_CPPFLAGS = \
 
 libatrildocument_la_CFLAGS = \
        $(LIBDOCUMENT_CFLAGS)                   \
-       $(SYNCTEX_CFLAGS)                       \
        $(WARN_CFLAGS)                          \
        $(DISABLE_DEPRECATED)                   \
        $(AM_CFLAGS)
 
+if ENABLE_SYNCTEX
+libatrildocument_la_CFLAGS += $(SYNCTEX_CFLAGS)
+endif
+
 libatrildocument_la_LDFLAGS = \
        -version-info $(EV_DOCUMENT_LT_VERSION_INFO)    \
        -no-undefined   \
@@ -116,9 +119,11 @@ libatrildocument_la_LDFLAGS = \
 libatrildocument_la_LIBADD = \
        $(LIBDOCUMENT_LIBS)     \
        $(GMODULE_LIBS)         \
-       $(SYNCTEX_LIBS)         \
        $(ZLIB_LIBS)
 
+if ENABLE_SYNCTEX
+libatrildocument_la_LIBADD += $(SYNCTEX_LIBS)
+endif
 
 BUILT_SOURCES =                        \
        ev-document-type-builtins.c     \
```
https://www.gnu.org/software/automake/manual/html_node/Usage-of-Conditionals.html

Reply:
I added your patch here too.